Bottom: a405aecb38de6061533b153ce54d09ba322d39d1
Top:    803709fd86524a8f481b31299cd912b1e707d7a9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 20:56:22 +0000

find.c: Fix inverted logic in LookupConnection()

When AndRats is true, we may consider starting with a ratline object,
otherwise we do no. I got this backwards by mistake.


---

diff --git a/src/find.c b/src/find.c
index 71e4b41..6fffb0d 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3087,7 +3087,7 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
   if (type == NO_TYPE)
     {
       type = SearchObjectByLocation (
-        LOOKUP_MORE & ~(AndRats ? RATLINE_TYPE : 0),
+        LOOKUP_MORE & ~(AndRats ? 0 : RATLINE_TYPE),
         &ptr1, &ptr2, &ptr3, X, Y, Range);
       if (type == NO_TYPE)
         return;
